common/short-hex-str/src/lib.rs

```diff
@@ -89,7 +90,13 @@ fn byte2hex(byte: u8) -> (u8, u8) {
 /// Hex encode a byte slice into the destination byte slice.
 #[inline(always)]
 fn hex_encode(src: &[u8], dst: &mut [u8]) {
-    debug_checked_precondition!(dst.len() == 2 * src.len());
+    debug_checked_precondition!(
+        dst.len()
+            == src
+                .len()
+                .checked_mul(2)
+                .expect("integer overflow when encoding to hex string")
```

Comment:
like I said, this is worse because it obscures the meaning of the assertion behind 5 lines of noise without actually gaining anything since it's behind a `debug_` which never runs in prod.

Reply:
OK, got it. will revert this file. Thanks!